data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/beacon/PostBlock.java

```diff
@@ -101,24 +119,69 @@ public void handleRequest(final RestApiRequest request) throws JsonProcessingExc
       return;
     }
 
-    final SignedBeaconBlock signedBeaconBlock = request.getRequestBody();
-
-    request.respondAsync(
-        validatorDataProvider
-            .submitSignedBlock(signedBeaconBlock)
-            .thenApply(
-                result -> {
-                  if (result.getRejectionReason().isEmpty()) {
-                    return AsyncApiResponse.respondWithCode(SC_OK);
-                  } else if (result
-                      .getRejectionReason()
-                      .get()
-                      .equals(FailureReason.INTERNAL_ERROR.name())) {
-                    return AsyncApiResponse.respondWithError(
-                        SC_INTERNAL_SERVER_ERROR, result.getRejectionReason().get());
-                  } else {
-                    return AsyncApiResponse.respondWithCode(SC_ACCEPTED);
-                  }
-                }));
+    Object requestBody = request.getRequestBody();
+
+    if (requestBody instanceof SignedBlockContents) {
+      request.respondAsync(
+          validatorDataProvider
+              .submitSignedBlockContents((SignedBlockContents) requestBody)
+              .thenApply(this::respond));
+
+    } else {
+      request.respondAsync(
+          validatorDataProvider
+              .submitSignedBlock((SignedBeaconBlock) requestBody)
+              .thenApply(this::respond));
+    }
+  }
```

Comment:
It looks like they both return a future of SendSignedBlockResult, might be a lot cleaner if we massage this a little...

```
final SafeFuture<SendSignedBlockResult> future = requestBody instanceOf SignedBlockContents ? validatorDataProvider.submitSignedBlockContents(requestBody) : validatorDataProvider
              .submitSignedBlock(requestBody);
 request.respondAsync(future.thenApply(this::respond);
 ```

Reply:
Done